### PR TITLE
Add path length option to sholl analysis

### PR DESCRIPTION
**Pull request by arnaudon**
By default the Sholl analysis is done wrt radial euclidean distance, but having path distance as an optional metric for Sholl analysis is sometimes useful. 
This PR provides such functionality via an additional parameters `distance_type`.

### CODE REVIEW

File: neurom/apps/cli.py
Comment (eleftherioszisis) [inline neurom/apps/cli.py:80]: What's different here?

Comment (arnaudon) [reply neurom/apps/cli.py:80]: coverage was complaining (but it wasn't in the previous PR...)

Comment (eleftherioszisis) [reply neurom/apps/cli.py:80]: You had introduce this change to fix the view cli in a previous commit. How does this change affect that issue you had before?

Comment (arnaudon) [reply neurom/apps/cli.py:80]: remember we discussed on slack that the issue in other PR was not about python3.7 as I naively assumed, but because there was not such option for 3d plots. So this is the correct way to implement the fix on the previous PR. (we discussed after it was merged I think). It re-appeared here for some weird reason that the try: except: was not tested (but it was in previous PR), so coverage was not at 100%. The end result is exactly the same.

Comment (eleftherioszisis) [reply neurom/apps/cli.py:80]: ok cool. I wanted to make sure all is good.

**Comment by eleftherioszisis** [comment]
LGTM. Maybe @andrisecker have some comments?

**Comment by arnaudon** [comment]
He is trying it, once he's happy with the result, we can merge!

File: neurom/features/morphology.py
Comment (eleftherioszisis) [inline neurom/features/morphology.py:503]: On second thought, this doesn't make much sense, does it?

Radial distance and path distance are two incompatible metrics. However, here are compared for the Sholl crossings, because you are using the radial distance of the shell against the path distance of the segments.

You may want to refer the the sholl descriptors in the following paper:
https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1010229

at the 2.2 section, point 7

> Total Wiring. Take a tree being a connected component of a neuron that contain soma, restricted to a ball of a radius r. This construction measures the total path length of this tree. This is the sum of the path distances of all dendritic segments and it is related to the notion of Arbor Density or Arbor Territory [[43](https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1010229#pcbi.1010229.ref043)]. Greater total wiring dendritic length means greater invaded space which render the dendrites available to more incoming connections and greater connectivity.

**Review by eleftherioszisis — CHANGES_REQUESTED** [review]

File: neurom/features/morphology.py
Comment (arnaudon) [reply neurom/features/morphology.py:503]: I'm not 100% sure I follow you, but now the 'shell' with radius `radius` is in path length, so it's not a spherical shell in 3d. I think this does what is in the paper you mention, no?

**Comment by arnaudon** [comment]
@andrisecker , are you good with this? Is it doing what you wanted?

File: neurom/features/morphology.py
Comment (eleftherioszisis) [reply neurom/features/morphology.py:503]: `radius` is not reflecting path length because the bins are calculated from the radial distance in the sholl frequency function:
```python
bins = np.arange(min_soma_edge, min_soma_edge + max(max_radius_per_neurite), step_size)
```
In the paper, it states indeed that you can have path distance descriptors with an independent var on the path length span L(N), and then they proceed to list the most common descriptors along with their physical interpretation. All of them use radial distance as the independent variable, including the total wiring I mentioned above that gives insight into arborization density.

Therefore, I wonder what the physical meaning of this descriptor is.

Comment (arnaudon) [reply neurom/features/morphology.py:512]: ah yes, good catch, I forgot to update the frequency bins, now it's consistent

**Comment by arnaudon** [comment]
@andrisecker are you still using this? do you think its worth merging or not?

**Comment by andrisecker** [comment]
@arnaudon I'm not using it any more, but I think if I had this use case and it's already implemented then you should merge it.

File: neurom/features/morphology.py
Comment (adrien-berchet) [inline neurom/features/morphology.py:561]:
```suggestion
        elif distance_type == "path":
```

**Comment by arnaudon** [comment]
@eleftherioszisis  I let you double check and approve this when you have time, as I cannot merge

File: neurom/features/morphology.py
Comment (arnaudon) [reply neurom/features/morphology.py:462]: done

File: neurom/features/morphology.py
Comment (arnaudon) [reply neurom/features/morphology.py:485]: I think its ok to use radius as a generic name, as it implies an underlying notion of distance, which can be whatever

**Comment by arnaudon** [comment]
rebase is a pain, and this feature is not used, I discard..

**Pull request closed without merging**
